EIPS/eip-5748.md

```diff
@@ -0,0 +1,191 @@
+---
+eip: 5748
+title: Approval Expiration for EIP-20 Tokens
+description: This standard introduces a standardized way for users to set expiration limits on ERC-20 token approvals.
+author: Go+ Security (@GoPlusSecurity), Jeff Hall (@Mr-Lucky), Xavi (@XaaaaavitheFool), Turan Vural (@turanzv)
+discussions-to: <https://ethereum-magicians.org/t/eip-5748-automatic-expiration-of-token-approvals/>
+status: Draft
+type: Standards Track
+category: Interface
+created: 2022-10-02) format
+requires:
+---
+
+This standard introduces a standard way for users to set expiration limits on ERC-20 token approvals to prevent the loss of assets due to contractrual vulnerabilities or authorization attacks.
+
+## Abstract
+This standard adds an expiration time to the `_approve()` function of token contracts to automatically recall approvals within a certain time period. This functionality as a standard will prevent vulnerabilty exploits due to outstanding token approvals.
+
+## Motivation
+As of the writing of this standard, the approval attack is the most common attack method in the industry with nearly all phishing attacks and asset theft related to it. Although good usage habits and basic knowledge of on-chain security can help users avoid a multitiude of exploits such as phishing, scams, DNS hijacking, JS injection, and other common approval attack methods, the spender's own security (or lack thereof) still needs to be fortified by strong code.
+
+In the Transit Swap attack of October 1st, 2022, hackers used a vulnerability in the Transit Swap contract to steal $20 million in assets. Due in part to professional security audits and the strong brand recognition of Transit Swap, many users were given a false sense of security and trusted the platform's service. Transit Swaps's contract had been running smoothly for over a year and had accumulated a large amount of user allownces, making the contract a high-reward target for malicious actors.
+
+There have been many similar incidents to the Transit Swap attack. Security audits cannot gurantee that a contract is 100% free from vulnerabilities. Some vulnerabilities are intentionally left by malicious developers themselves. In line with good trust practices and general security hygeine, unconditional trust is risky, and modifying approvals into limited trusts with automatic recovery will be a key tool to natively prevent attacks.
+
+## Specification
+
+A time period `_expiration` is mapped on top of the original `allowance` to specify an approval time limit.
+
+The internal  `_approve()` method adds a uint256 `period` to the entry and stored in `_expiration`.
+
+A `DEFAULT_EXPIRATION` constant is added for implementations that do not specify the `period` parameter. A getter method `defaultExpiration()` is included in the implementation for transparency to the user in the approval process.
```

Comment:
Your tone here is a bit... off. You're writing your proposal like you're walking someone through your decision process. While there is a place for this type of information (the `Rationale` section), the `Specification` section should clearly state the requirements for a contract to meet this standard. Perhaps something like:

```suggestion
Contracts implementing this standard MUST expose a `DEFAULT_EXPIRATION` constant of type `uint256`. This constant stores the expiration period to use for `approve` calls that do not include a `period` argument.

`approve` functions without a `period` argument MUST use the expiry period exposed in the `DEFAULT_EXPIRATION` constant.
```